source/material_model/rheology/visco_plastic.cc

```diff
@@ -256,12 +257,17 @@ namespace aspect
             // Step 3b: calculate current (viscous or viscous + elastic) stress magnitude
             double current_stress = 2. * viscosity_pre_yield * current_edot_ii;
 
-            // Step 4: calculate strain-weakened friction, cohesion
+            // Step 4a: calculate strain-weakened friction, cohesion
             const DruckerPragerParameters drucker_prager_parameters = drucker_prager_plasticity.compute_drucker_prager_parameters(j,
                                                                       phase_function_values,
                                                                       n_phases_per_composition);
             const double current_cohesion = drucker_prager_parameters.cohesion * weakening_factors[0];
-            const double current_friction = drucker_prager_parameters.angle_internal_friction * weakening_factors[1];
+            output_parameters.current_friction_angles[j] = drucker_prager_parameters.angle_internal_friction * weakening_factors[1];
+
+            // Steb 4b: calculate friction angle dependent on strain rate if specified
+            if (friction_options.get_friction_dependence_mechanism() == dynamic_friction)
+              output_parameters.current_friction_angles[j] = friction_options.compute_dependent_friction_angle(current_edot_ii,
+                                                             j, output_parameters.current_friction_angles[j]);
```

Comment:
Is there a preferred order in which the changes to the friction angle should be applied. E.g. first compute the rate-dependent friction and then apply the strain weakening?

Reply:
that is a very good question, thanks for spotting this! I am actually not sure whether there is a preferred order or not. I tend to say that rate-and-state should come first and then strain weakening? Does someone else have a preference or even an explanation ready?

Reply:
I put a comment into the code now, stating, that I dont know the right order....